ibm/mas_devops/roles/ocp_provision/tasks/providers/roks.yml

```diff
@@ -110,7 +110,61 @@
       {{ roks_flags }}
 
 
-# 8. Watch cluster provisioning progress
+# 8. Add a GPU worker node if ocp_provision_gpu is true
+# -----------------------------------------------------------------------------
+- name: "roks : Check if GPU worker pool already exists"
+  when:
+    - ocp_provision_gpu is defined
+    - ocp_provision_gpu|bool == true
+  shell: ibmcloud oc worker-pool get --cluster {{ cluster_name }} --worker-pool gpu --output json
+  register: workerpool_lookup
+  failed_when: "workerpool_lookup.rc > 1"
+
+- name: "roks : Debug worker pool status"
+  when:
+    - ocp_provision_gpu is defined
+    - ocp_provision_gpu|bool  == true
+    - workerpool_lookup.rc == 0
+  debug:
+    msg: "A gpu worker pool already exists, skipping creation of gpu worker pool"
+
+- name: "roks : Add an additional GPU worker pool"
+  when:
+    - ocp_provision_gpu is defined
+    - ocp_provision_gpu|bool  == true
+    - workerpool_lookup.rc == 1
+  shell: >
+    ibmcloud oc worker-pool create classic \
+      --hardware dedicated \
+      --cluster {{ cluster_name }} \
+      --name gpu \
+      --flavor mg4c.32x384.2xp100 \
+      --size-per-zone {{ gpu_workers }} \
+      --entitlement cloud_pak
+
+- name: "roks : Add GPU worker pool to zone"
+  when:
+    - ocp_provision_gpu is defined
+    - ocp_provision_gpu|bool  == true
+    - workerpool_lookup.rc == 1
```

Comment:
It looks like we only do this task if the workerpool lookup earlier reported that it was already present, but it could be present but not added to the zone (due to an error after the worker-pool create). So next time it is called it doesn't add the worker to the zone. Is there harm in just calling `ibmcloud oc zone add` if it is already added?

Reply:
Yes, there is an error if the worker-pool has already been added to the zone. I am going to add a part to check if it has been added or not before adding. Maybe that would solve this problem

Reply:
changed made and tested